flytekit/clients/raw.py

```diff
@@ -63,16 +63,24 @@ def _refresh_credentials_basic(flyte_client):
     :param flyte_client: RawSynchronousFlyteClient
     :return:
     """
-    auth_endpoints = _credentials_access.get_authorization_endpoints(flyte_client.url)
-    token_endpoint = auth_endpoints.token_endpoint
-    client_secret = _basic_auth.get_secret()
-    cli_logger.debug(
-        "Basic authorization flow with client id {} scope {}".format(_CLIENT_ID.get(), _get_basic_flow_scopes())
-    )
-    authorization_header = _basic_auth.get_basic_authorization_header(_CLIENT_ID.get(), client_secret)
-    token, expires_in = _basic_auth.get_token(token_endpoint, authorization_header, _get_basic_flow_scopes())
+    if not flyte_client.oauth2_metadata or not flyte_client.public_client_config:
+        raise ValueError(
+            "Raw Flyte client attempting client credentials flow but no response from Admin detected. "
+            "Check your Admin server's .well-known endpoints to make sure they're working as expected."
+        )
+
+    token_endpoint = flyte_client.oauth2_metadata.token_endpoint
+    scopes = flyte_client.oauth2_metadata.scopes_supported
```

Comment:
For public client case, we should request scopes listed [here](https://github.com/flyteorg/flyteidl/blob/master/protos/flyteidl/service/auth.proto#L55) instead... the supported scopes can be a huge list...

For client_credentials flow, users should be able to set that just like they set client id and client_secret... if not passed, we default to the public client's scopes [here](https://github.com/flyteorg/flyteidl/blob/0ff9b16ad4c37971dfaa1e51b5fe970bae40e356/clients/go/admin/token_source_provider.go#L140).